packages/kbn-test/src/functional_tests/lib/run_kibana_server.js

```diff
@@ -1,17 +1,37 @@
+import { resolve } from 'path';
 import { KIBANA_ROOT, KIBANA_EXEC, KIBANA_EXEC_PATH } from './paths';
 
-export async function runKibanaServer({ procs, config }) {
-  const cliArgs = config.get('kibanaServerArgs') || [];
+export async function runKibanaServer({ procs, config, options }) {
+  const { installDir } = options;
 
-  // start the kibana server and wait for it to log "Server running" before resolving
   await procs.run('kibana', {
-    cmd: KIBANA_EXEC,
-    args: [KIBANA_EXEC_PATH, ...cliArgs],
+    cmd: getKibanaCmd(installDir),
+    args: getCliArgs(config, installDir),
     env: {
       FORCE_COLOR: 1,
       ...process.env,
     },
-    cwd: KIBANA_ROOT,
+    cwd: installDir || KIBANA_ROOT,
     wait: /Server running/,
   });
 }
+
+function getKibanaCmd(installDir) {
+  if (installDir) {
+    return process.platform.startsWith('win')
+      ? resolve(installDir, 'bin/kibana.bat')
+      : resolve(installDir, 'bin/kibana');
+  }
+
+  return KIBANA_EXEC;
+}
+
+function getCliArgs(config, installDir) {
+  const buildArgs = config.get('kbnTestServer.buildArgs') || [];
+  const sourceArgs = config.get('kbnTestServer.sourceArgs') || [];
+  const serverArgs = config.get('kbnTestServer.serverArgs') || [];
+
+  return installDir
+    ? [...serverArgs, ...buildArgs]
+    : [KIBANA_EXEC_PATH, ...serverArgs, ...sourceArgs];
+}
```

Comment:
Here's where we collect the right CLI args based on whether installDir was passed (based `--kibana-install-dir`)